src/libraries/Common/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -479,13 +481,13 @@ internal static Exception InvalidDataLength(long length)
 
         internal static bool IsEmptyArray(string[] array) => (null == array) || (0 == array.Length);
 
-        internal static bool IsNull(object value)
+        internal static bool IsNull(object? value)
```

Comment:
I don't know if it would help anything, but the argument could be attributed as `[NotNullWhen(false)]`